src/nupic/types/Fraction.cpp

```diff
@@ -366,7 +366,7 @@ namespace nupic
 
     while(components.size())
     {
-      std::swap<int>(numerator_, denominator_);
+      std::swap(numerator_, denominator_);
```

Comment:
The variables being swapped aren't pointers, they are of type `int`, so I wonder if we actually want to call swap on the references:
```
std::swap<int>(&numerator_, &denominator_);
```

It looks like there are tests [here](https://github.com/numenta/nupic.core/blob/82296d52f7676f2bd00ddf4e295acd28ce7f6ce6/src/test/unit/types/FractionTest.cpp#L319) so we should be safe as long as those pass.

But I really don't think we want to remove the template param because then we really don't know if the right thing is happening.